### PR TITLE
[SPARK-47828][CONNECT][PYTHON][3.4] DataFrameWriterV2.overwrite fails with invalid plan

### PR DESCRIPTION
**Pull request by zhengruifeng**
cherry pick fix https://github.com/apache/spark/pull/46023 to 3.4

### CODE REVIEW

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — COMMENTED** [review]
Is this a regression, @zhengruifeng and @HyukjinKwon ?

**Review by dongjoon-hyun — COMMENTED** [review]
BTW, please rebase this, @zhengruifeng .

In branch-3.4, `build/mvn versions:set` seems to be quite unstable. It causes two wrong commits inside the release script.

File: python/pyspark/sql/connect/plan.py [hunk 1]
```diff
@@ -1544,16 +1544,16 @@ def command(self, session: "SparkConnectClient") -> proto.Command:
...
             elif wm == "overwrite_partition":
                 plan.write_operation_v2.mode = proto.WriteOperationV2.Mode.MODE_OVERWRITE_PARTITIONS
             elif wm == "append":
                 plan.write_operation_v2.mode = proto.WriteOperationV2.Mode.MODE_APPEND
             elif wm == "replace":
                 plan.write_operation_v2.mode = proto.WriteOperationV2.Mode.MODE_REPLACE
-                if self.overwrite_condition is not None:
-                    plan.write_operation_v2.overwrite_condition.CopyFrom(
-                        self.col_to_expr(self.overwrite_condition, session)
-                    )
```
Comment (dongjoon-hyun) [inline python/pyspark/sql/connect/plan.py:1556]: According to the commit history, this is a bug of [SPARK-42002](https://issues.apache.org/jira/browse/SPARK-42002) from Apache Spark 3.4.0. Not a regression of 3.4.1~3.4.3. Did I understand correctly?

**Review by dongjoon-hyun — COMMENTED** [review]
Since this is a bug of the original Apache Spark 3.4.0 due to [SPARK-42002](https://issues.apache.org/jira/browse/SPARK-42002) and SPARK-47828 wasn't marked as blockers.

If you don't mind, shall we wait for a few days until the on-going vote finish,@zhengruifeng and @HyukjinKwon ?

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
I'll block this for the further discussion and conclusion.

**Comment by HyukjinKwon** [comment]
It's not a blocker but just a bug fix. Yeah let's wait it out 👍

**Comment by dongjoon-hyun** [comment]
Thank you!

File: python/pyspark/sql/connect/plan.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/connect/plan.py:1556]: yes, it is a bug fix. Not a regression

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @zhengruifeng and @HyukjinKwon 
Merged to branch-3.4.

**Pull request closed without merging**